velox/exec/tests/AggregationTest.cpp

```diff
@@ -1245,6 +1245,91 @@ TEST_F(AggregationTest, groupingSets) {
       "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
 }
 
+const std::shared_ptr<const core::GroupIdNode> findGroupIdNode(
+    const core::PlanNodePtr& root,
+    const core::PlanNodeId& groupIdNodeId) {
+  std::vector<core::PlanNodePtr> nodes;
+  nodes.push_back(root);
+  while (!nodes.empty()) {
+    std::vector<core::PlanNodePtr> nextNodes;
+    for (const auto& node : nodes) {
+      if (node->id() == groupIdNodeId) {
+        return std::dynamic_pointer_cast<const core::GroupIdNode>(node);
+      }
+      const auto childNodes = node->sources();
+      std::copy(
+          childNodes.begin(), childNodes.end(), std::back_inserter(nextNodes));
+    }
+    nodes.swap(nextNodes);
+  }
+  VELOX_UNREACHABLE("Plan node {} is not found", groupIdNodeId);
+  return nullptr;
+}
+
+TEST_F(AggregationTest, groupingSetsOutput) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<StringView>(
+              size,
+              [](auto row) { return StringView(std::string(row % 12, 'x')); }),
+      });
+
+  createDuckDbTable({data});
+
+  core::PlanNodeId randomGroupId;
+  core::PlanNodeId orderGroupId;
+  auto randomPlan =
+      PlanBuilder()
+          .values({data})
+          .groupId({{"k2", "k1"}, {}}, {"a", "b"})
+          .capturePlanNodeId(randomGroupId)
+          .singleAggregation(
+              {"k2", "k1", "group_id"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
```

Comment:
Do you need this project?

Reply:
When compare the result of two plan, we may need this project to ensure the same output.